deployments/helm/dra-example-driver/templates/kubeletplugin.yaml

```
@@ -12,10 +12,18 @@ spec:
     matchLabels:
       {{- include "dra-example-driver.selectorLabels" . | nindent 6 }}
       app.kubernetes.io/component: kubeletplugin
+  {{- if .Values.kubeletPlugin.seamlessUpgrades.enabled }}
+  updateStrategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxSurge: 1
+      maxUnavailable: 0
+  {{- else }}
```

Setting `kubeletPlugin.seamlessUpgrades.enabled` will silently ignore the `kubeletPlugin.updateStrategy` value which isn't ideal.

Maybe we could keep the existing `updateStrategy` template block and use a different default for `maxSurge` and `maxUnavailable` when seamless upgrades are enabled? Then `kubeletPlugin.updateStrategy` is still authoritative if users express an opinion on that, but only setting `kubeletPlugin.seamlessUpgrades.enabled` is enough to set them to values that actually exercise seamless upgrades if they don't.

---

With my changes the user choice always wins so if we have seamless upgrades enabled and maxSurge/maxUnavailable specified then it would honor that, which wouldn't work with seamlessUpgrades

---

Just to confirm, if user provides updateStrategy but seamlessUpgrades is enabled, then overwrite just the maxSurge and maxUnavailable of the given updateStrategy?